doc/adr/0011-tsp-analysis-api.md

```diff
@@ -0,0 +1,268 @@
+
+### Global analysis service
+
+An analysis service for managing global analyses will be introduced. Global in this context means that the analysis definitions will be handled on the application level. 
+
+    GET /analysis/
+        returns a list of analysis type descriptors: typeId, name, description, expected input parameter list ("path" for file path)
+    GET /analysis/types
+        returns a map typeId -> list of analysis descriptor of existing analyses on server: analysisId, name, description, parameters
+    GET /analysis/types/{typeId}
+        returns a list of analysis descriptors for given typeId
+    POST /analysis/types/{typeId}
+        Upload an analysis definition for given typeId
+        Returns a new analysis descriptor with unique analysisID
+    PUT /analysis/types/{typeId}/{analysisId}
+        Update an analysis
+    GET /analysis/types/{typeId}/{analysisId}
+        Returns the analysis descriptor for given analysisId
+    DELETE /analysis/types/{typeId}/{analysisId}
+        Delete an analysis
+
+#### Analysis type descriptor
+
+The analysis type descriptor describes the type of external analysis. Different types have different syntax and hence back-end implementation. This descriptor is used to distinquish the different external analysis. The trace server implementation will provide the list of type descriptors on client requests.
+```javascript
+AnalysisTypeDescriptor {
+    name: string,
+    description: string,
+    id: string,
+    scope: string,
+    inputParameter: string[]
+}
+```
+
+Where:
+
+- `name`: Name of the analysis type. Can be shown to the end-user.
+- `description`: The description of the analysis type. Can be shown to the end-user.
+- `id`: Unique id of the analysis type. Used in the application to distinquish analysis types
+- `scope:` `experiment` for analysis per experiment or `global` for all experiments
+- `inputParameter`: A list of input paramter that the front-end needs to provide. Use "path" for file path. Only path supported as the first iteration.
+
+#### Analysis descriptor
+
+The analysis descriptor describes an instance of an external analysis for a given type. This descriptor is used to distinquish the different external analysis for a given type.
+
+```javascript
+AnalysisDescriptor {
+    name: string,
+    description: string,
+    id: string,
+    typeId: string
+    parameters: map<string, object>
+}
+```
+
+Where:
+- `name`: Name of the analysis type. Can be shown to the end-user.
+- `description`: The description of the analysis type. Can be shown to the end-user.
+- `id`: Unique id of the analysis type. Used in the application to distinquish analysis types
+- `typeId`: ID of the analysis type.
+- `parameter`: A map of paramter corresponding to the `inputParameter` specified in the analysis type descriptor
```

Comment:
parameter

Reply:
> To be clear, is this a map where the key is the parameter name and the value(object) is the parameter value?

yes, so for the example of files, the key will be `path` and the value the actual file path.